### PR TITLE
Add DSA device plugin.

### PR DESCRIPTION
**Pull request by pingzhaozz**
Disable by default. Update redis/values.yaml to enable it.

### CODE REVIEW

File: k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{{- if .Values.dsa.enabled }}
```
Comment (leyao-daily) [inline k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml]: I think it won't work here.

File: k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          path: /dev/char
+      - name: sysfs
+        hostPath:
+          path: /sys/bus/dsa
+      - name: kubeletsockets
+        hostPath:
+          path: /var/lib/kubelet/device-plugins
+      nodeSelector:
+        kubernetes.io/arch: amd64
+{{- end }}
```
Comment (leyao-daily) [inline k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml]: Add new line at the end.

File: helm/redis/values.yaml [hunk 3]
```diff
@@ -43,6 +44,9 @@ resources: {}
   #   cpu: 100m
   #   memory: 128Mi
 
+dsa:
+  #enabled: true
```
Comment (leyao-daily) [inline helm/redis/values.yaml]: maybe we define this as false defautly.

File: helm/redis/values.yaml [hunk 4]
```diff
@@ -5,6 +5,7 @@
 replicaCount: 1
 
 redisImage:
+  #repository: <your registry>
```
Comment (leyao-daily) [inline helm/redis/values.yaml]: Remove the comment. And for redis dsa image, which we build, i think we should same with others. Just add a new field:
```
image:
  repository: <your registry>
  pullPolicy: IfNotPresent
  # Overrides the image tag whose default is the chart appVersion.
  tag: ""
```

And remove the duplicated pull policy from redisImage.

File: helm/redis/templates/deployment.yaml [hunk 5]
```diff
@@ -30,12 +30,22 @@ spec:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
       containers:
         - name: {{ .Chart.Name }}
-          securityContext:
-            {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- if .Values.dsa.enabled }}
+          image: "{{ .Values.redisImage.repository }}/cnap-redis:{{ .Values.redisImage.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.redisImage.pullPolicy }}
```
Comment (leyao-daily) [inline helm/redis/templates/deployment.yaml]: Move this out of the if statement,

File: k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml (hunk 2 above)
Comment (pingzhaozz) [reply k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml]: ok

File: helm/redis/values.yaml (hunk 3 above)
Comment (pingzhaozz) [reply helm/redis/values.yaml]: ok

File: helm/redis/values.yaml (hunk 4 above)
Comment (pingzhaozz) [reply helm/redis/values.yaml]: combine the redisimage and image field

File: k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml (hunk 1 above)
Comment (pingzhaozz) [reply k8s-manifests/plugin/dsa/intel-dsa-plugin.yaml]: ok, removed.

**Review by leyao-daily — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**